api/proto/teleport/userintegrationtasks/v1/user_integration_tasks.proto

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.userintegrationtasks.v1;
+
+import "google/protobuf/timestamp.proto";
+import "teleport/header/v1/metadata.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/userintegrationtasks/v1;userintegrationtasksv1";
+
+// UserIntegrationTask is a resource that represents an action to be completed by the user.
+// UserIntegrationTasks are a unit of work for users to act upon issues related to integrations.
+// As an example, when auto-enrolling EC2 instances using the Discovery Service
+// a UserIntegrationTask is created to let the user know that something failed on a set of instances.
+// The user can then mark the task as resolved after following the recommendation/fixing steps.
+message UserIntegrationTask {
+  // The kind of resource represented.
+  string kind = 1;
+  // Mandatory field for all resources. Not populated for this resource type.
+  string sub_kind = 2;
+  // The version of the resource being represented.
+  string version = 3;
+  // Common metadata that all resources share.
+  teleport.header.v1.Metadata metadata = 4;
+  // The configured properties of UserIntegrationTask.
+  UserIntegrationTaskSpec spec = 5;
+}
+
+// UserIntegrationTaskSpec contains the properties of the UserIntegrationTask.
+message UserIntegrationTaskSpec {
+  // Integration is the integration name that originated this task.
+  string integration = 1;
+  // TaskType indicates the type of task.
+  // Examples: discover-ec2, discover-rds, discover-eks
+  string task_type = 2;
+  // IssueType is an identifier for the type of issue that happened.
+  // Example for discover-ec2: SSM_AGENT_NOT_AVAILABLE
+  string issue_type = 3;
```

Comment:
Should this be an enum or is this a free-form field?

Reply:
I prefer to keep it as a string instead with well-known values. It's easier for evolving and to keep IaC tools happy.